manim/mobject/geometry/shape_matchers.py

```diff
@@ -87,19 +90,16 @@ def construct(self):
 
     def __init__(
         self,
-        mobject: Mobject,
+        *mobjects,
         color: ParsableManimColor | None = None,
         stroke_width: float = 0,
         stroke_opacity: float = 0,
         fill_opacity: float = 0.75,
         buff: float = 0,
         **kwargs,
     ):
-        if color is None:
```

Comment:
It seems that checking `type(mob) == Mobject`, which I included as a typecheck by your suggestion, creates a circular input, because one of the methods of Mobject, `add_background_rectangle`, instantiates a BackgroungRectangle. Despite the fact that Mobject is never instantialised, the mere reference to it outside of argument typing requires an import, entering the dependency loop.
I am at a loss as to what to do here, because it seems that this circular import can only be solved by removing one of those features. I still haven't learned of a way to resolve cyclic imports in python.
I would really appreciate it if you had a solution to this problem, because I have come across this issue in some of my personal projects, and always had to add workarounds to resolve it.

Reply:
Adding on top of that, there's no way that the import of Group induced this issue, because the class is literally 3 lines long and immediately calls back to the super constructor.
![Uploading image.png…]()

Reply:
One thing you could try is moving the import of `BackgroundRectangle` to be inside of the `add_background_rectangle` method.

Reply:
Thanks, but that is already the case.
By excluding literally everything else, I found that moving the import of Group into the body of SurroundingRectangle does it. That's really weird, because the import of Mobject at the top of shape_matchers.py doesn't seem to cause the same issues.
Regardless, it works now.

P.S.: Sorry about the truckload of commits, it took me a while to figure out the solution. It was literally just trial-and-error.